pkg/controller/machine/controller.go

```diff
@@ -422,21 +422,26 @@ func isInvalidMachineConfigurationError(err error) bool {
 func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, errorMessage string) error {
 	if stringPointerDeref(machine.Status.Phase) != phase {
 		klog.V(3).Infof("%v: going into phase %q", machine.GetName(), phase)
+		// We are required to Patch annotations before updating the status because Machines have a status subresource.
+		// If we modify the status before Patching the annotations, the API server will ignore the status updates and once
+		// the request is returned, the client will restore the `machine` object to the state as the API server sees it,
+		// overwriting any status changes we had already made.
+		if phase == phaseFailed {
+			err := r.patchFailedMachineInstanceAnnotation(machine)
+			if err != nil {
+				klog.Errorf("Failed to update machine %q: %v", machine.GetName(), err)
+				return err
+			}
+		}
+
+		// Patch status
```

Comment:
"Since we may have mutated the local copy of the machine above, we need to calculate baseToPatch here."